components/DaoMenu.tsx

```diff
@@ -0,0 +1,81 @@
+import { Fragment } from 'react';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import clsx from 'clsx';
+import { Menu, Transition } from '@headlessui/react';
+import { ChevronDownIcon } from '@heroicons/react/20/solid';
+import { useConfig } from '@/hooks/useConfig';
+import { DaoId } from '@/config';
+
+type Option<T> = {
+  value: T;
+  label: string;
+  logo: string;
+};
+
+type Props<T> = {
+  options: Option<T>[];
+};
+
+const DaoMenu = <T,>(props: Props<T>) => {
```

Comment:
Curious why we're using generics here

Reply:
I used it mainly to give us flexibility if we decided to change types for the value. Right now, we are using numbers, but like you suggested we might want to switch to strings.